### PR TITLE
fix: Onboarding steps selection uncheck

### PR DESCRIPTION
**Pull request by anastasiapyzhik**
https://artsyproduct.atlassian.net/browse/GRO-248

This fixes the behavior of artists and categories checking within the onboarding steps.

**Before**
before

**After**
after

### CODE REVIEW

File: src/v2/Components/Onboarding/GenesStep/GenesStep.tsx [hunk 1]
```diff
@@ -35,11 +35,11 @@ export const GenesStep: React.FC<Props> = props => {
     leading: true,
   })
 
-  const handleGeneFollow = (newCount, gene) => {
-    setFollowCount(newCount)
+  const handleGeneFollow = (follow, gene) => {
+    setFollowCount(state => (follow ? state + 1 : state - 1))
```
Comment (jonallured) [inline src/v2/Components/Onboarding/GenesStep/GenesStep.tsx:39]:
```suggestion
    setFollowCount(previousCount => (follow ? previousCount + 1 : previousCount - 1))
```

File: src/v2/Components/Onboarding/ArtistsStep/ArtistsStep.tsx [hunk 2]
```diff
@@ -36,11 +36,13 @@ export const ArtistsStep: React.FC<Props> = props => {
     leading: true,
   })
 
-  const handleArtistFollow = (newCount, artist) => {
-    setFollowCount(newCount)
+  const handleArtistFollow = (follow, artist) => {
+    setFollowCount(state => (follow ? state + 1 : state - 1))
```
Comment (jonallured) [inline src/v2/Components/Onboarding/ArtistsStep/ArtistsStep.tsx:40]:
```suggestion
    setFollowCount(previousCount => (follow ? previousCount + 1 : previousCount - 1))
```

File: src/v2/Components/Onboarding/GenesStep/GeneSearchResults.tsx [hunk 3]
```diff
@@ -43,7 +43,6 @@ const NoResultsContainer = styled.div`
 
 class GeneSearchResultsContent extends React.Component<Props, null> {
   private excludedGeneIds: Set<string>
-  followCount: number = 0
```
Comment (jonallured) [inline src/v2/Components/Onboarding/GenesStep/GeneSearchResults.tsx:46]: I really like how you were able to remove these properties from the classes - every little bit of simplification helps! 🏆

File: src/v2/Components/Onboarding/ItemLink.tsx [hunk 4]
```diff
@@ -48,6 +48,10 @@ interface Props extends React.HTMLProps<HTMLAnchorElement> {
   id: string
   name: string
   image_url: string
+  onFollow: (
+    e: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
```
Comment (jonallured) [inline src/v2/Components/Onboarding/ItemLink.tsx:52]: Question: does this type need to be tied to the mouse? I wonder if there's a more general type that would also work? 🤔

File: src/v2/Components/Onboarding/ItemLink.tsx [hunk 5]
```diff
@@ -64,8 +68,14 @@ export default class ItemLink extends React.Component<Props, State> {
...
 
   onClick(e) {
-    this.props.onClick(e)
-    this.setState({ selected: true })
+    this.setState(
+      state => ({
+        selected: !state.selected,
+      }),
+      () => {
+        this.props.onFollow(e, this.state.selected)
```
Comment (jonallured) [inline src/v2/Components/Onboarding/ItemLink.tsx:76]: This turned out real nice 👍

**Comment by jonallured** [comment]
Oh oops, I hit submit on that review too quickly! 😝 

I meant to also bring up tests: I wonder if there's something we can do here to update our tests to reflect this new behavior - any ideas @anastasiapyzhik ?

File: src/v2/Components/Onboarding/ItemLink.tsx (hunk 4 above)
Comment (anastasiapyzhik) [reply src/v2/Components/Onboarding/ItemLink.tsx:52]: I just left the initial type for click event as the 'onClick' method passes its event to the 'onFollow' method. But I've just realized that 'onFollow' method doesn't use the event object at all and I should've removed it from parameters.

**Comment by anastasiapyzhik** [comment]
@jonallured, yeah, I think we can check the `ItemLink` component for the rendering the proper icon and also I'll investigate how I can test `ArtistsStep` and `GenesStep` components. Probably I'll make something similar to `CollectorIntent` component testing.

**Comment by anastasiapyzhik** [comment]
@artsy/grow-devs I've got a question. Do I understand it right that onboarding steps are available to users only when signing up and there is no need to remember which artists/categories are already followed by a user when opening onboarding steps?

**Comment by The-Beez-Kneez** [comment]
@anastasiapyzhik as far as I am aware the onboarding steps are only available when signing up. And a user has no previously followed categories, as they are new. (@jonallured correct me if I am wrong on that, you been in onboarding recently).

**Review by The-Beez-Kneez — APPROVED** [review]

**Review by damassi — APPROVED** [review]
Looks great @anastasiapyzhik - once a the tests are complete and the PR is 🍏 feel free to merge!

**Comment by jonallured** [comment]
Yep, what @The-Beez-Kneez said is correct! 👍

**Comment by The-Beez-Kneez** [comment]
Would also like to second the need for some testing surrounding this functionality (as it could have caught it originally). Also need to make sure we update any existing tests that may be reacting to this new functionality with failures.

But otherwise, looks great!

**Comment by anastasiapyzhik** [comment]
@The-Beez-Kneez could you clarify what do you mean by testing surrounding the functionality? Adding more tests not directly related to the functionality?

**Comment by anastasiapyzhik** [comment]
@artsy/grow-devs remind me please, what should I do once I merged PR? Should I ping someone?

**Comment by damassi** [comment]
@anastasiapyzhik, once PR is merged you can move your JIRA ticket into the next column (`merged`); thats how we track things! No need to ping.

**Pull request merged**